### DISCUSSION TITLE
C++17 <filesystem> header (GCC 8+)

### DISCUSSION THREAD

**Discussion by AbitTheGray** [community/community#25473] [Actions]
<p>First time attempting to use Github Actions with my C++20 project.<br>
Got it all working until it reached <code>#include &lt;filesystem&gt;</code> which is C++17 standard header.</p>
<p>GCC should support C++17 from version 8 but <code>ubuntu-latest</code> seems to use <code>GNU 7.5.0</code> which was released in November 2019.<br>
<strong>Is there a way to make it use newer version?</strong> Or do I need to wait for someone from Github to update it?</p>
<ul>
<li><a href="https://github.com/AbitTheGray/Cpp-LUA-Template/runs/1235495568" rel="noopener nofollow ugc">Action log</a></li>
<li><code>runs-on: ubuntu-latest</code></li>
<li>
<a href="https://github.com/AbitTheGray/Cpp-LUA-Template/blob/0a160f70bbb55bb2766653d6f0be2ffd09895b4e/.github/workflows/cmake.yml" rel="noopener nofollow ugc">Action Config</a>
<ul>
<li>Original Action: <code>actions/starter-workflows/</code>, <code>/ci/cmake.yml</code>
</li>
</ul>
</li>
</ul>
<p>PS: Tried to link everything but it does allow me only 2 links so you must find it yourself.</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>According to the <a href="https://github.com/actions/virtual-environments/blob/65f9b7ff6b853255fda580461643b3b239b17250/images/linux/Ubuntu1804-README.md">virtual environment documentation for <code>ubuntu-latest</code></a> (currently still at 18.04) it has a bunch of GCC versions installed:</p>
<blockquote>
<ul>
<li>GNU C++ 7.5.0, 8.4.0, 9.3.0, 10.1.0</li>
</ul>
</blockquote>
<p>If they follow the usual Ubuntu and Debian GCC packaging conventions you should be able to select the major version by suffixing <code>gcc</code> command with the version, e.g. run <code>gcc-10</code> to select GCC 10.</p>
<p>Alternatively you can use <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idcontainer"><code>jobs.&lt;job_id&gt;.container</code></a> and use a container image that contains (or lets you install) whichever version you need.</p>
